Add possiblity for Domain-U creation in AoS Zephyr Dom0

firscity opened the pull request.

xakep-amatop commented on src/domains/domu/domu_cfg.c:
```diff
@@ -0,0 +1,89 @@
+        }
+    }
+
+    rc = domain_create(&domu_cfg, 2);
```
as far as I remember we should about usage of static dom IDs
and we have some code for getting Dom IDs

firscity replied:
Code for receiving domid was added right after previous push. This version is now using it for creation.

firscity commented on src/domains/dom_runner.c:
```diff
@@ -0,0 +1,37 @@
+{
+    int rc;
+
+    rc = domain_create(&domd_cfg, 1);
```
Sure, some backends in generic case may be located e.g. in Dom0, but since it is a product with some static configuration without such cases, I prefer to left it as it was.

firscity commented on src/domains/domu/domu_cfg.c:
```diff
@@ -0,0 +1,89 @@
+    .gnt_frames          = 32,
+    .max_maptrack_frames = 1,
+
+    .gic_version = XEN_DOMCTL_CONFIG_GIC_V3,
```
Thank you for catch, added `#ifdef` for this.

firscity commented:
Since last push:
- Added DOMID_DOMD define and check if created domain received it
- Fixed GIC version for DomU
- Added correct handling for domid after zephyr-xenlib was updated

al1img commented on Kconfig:
```diff
@@ -104,4 +104,17 @@ config DOMD_DTB_PATH
 config TA_DEPLOY_DIR
 	string "Location of optee TA's deploy dir"
 
+config AOS_ENABLE_DOMU
```
Not directly related to AOS. `AOS_` prefix can be removed.

firscity replied:
This config will be exported globally over the project it should have some specific naming to prevent collision. Since now it affect only AoS, I used such naming. Can you suggest anything else?

al1img replied:
`DOMU_..` is good prefix in this case: `DOMU_ENABLE`. Looks good for me.

GrygiriiS commented on src/domains/domu/domu_runner.h:
```diff
@@ -0,0 +1,18 @@
+extern "C" {
+#endif
+
+int domu_start();
```
Logically it has to be called somewhere under CONFIG_AOS_ENABLE_DOMU ifdef.

GrygiriiS replied:
Sry. missed it

lorc replied:
`int domu_start(void);` probably. Looking at `#ifdef __cplusplus` you expect that this code *might* be compiled with plain C, so it should conform plain C declaration.

GrygiriiS commented on src/domains/domu/domu_cfg.c:
```diff
@@ -0,0 +1,94 @@
+struct xen_domain_cfg domu_cfg = {
+    .mem_kb            = 0x40000, /* 256Mb */
+
+    .flags               = (XEN_DOMCTL_CDF_hvm | XEN_DOMCTL_CDF_hap | XEN_DOMCTL_CDF_iommu),
```
XEN_DOMCTL_CDF_iommu - Are you sure it's needed for DomU? RPI5 for example will just fail with this flag set.

lorc replied:
I believe it is not needed in this particular case, because this DomU has no dtdevs assigned for it.  On other hand, having this option will not break anything

GrygiriiS reviewed (commented):
One comment. otherwise ` Reviewed-by: Grygorii Strashko <grygorii_strashko@epam.com>`

firscity commented:
Added Grygoriy's RB, thanks for review.

lorc reviewed (approved):
With that small nit fixed:

`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

firscity commented:
Last push:
- rebased on top of aoscloud/develop
- added `(void)` to function declaration
- added Volodymyr AB

firscity commented:
Last push:
Changed `CONFIG_AOS_ENABLE_DOMU` -> `CONFIG_DOMU_ENABLE`

firscity commented:
Last push:
Fixed formatting via clang-format
